### ISSUE TITLE
define/implement exhaustiveness checking of type-aliased enums

### ISSUE THREAD

**Issue by nishanths** [nishanths/exhaustive#13]

**Comment by psucodervn** [comment]
Hi, any update on this?

**Comment by nishanths** [comment]
Hi, I'm working on this now. It expect it should be ready in around a week.

**Comment by nishanths** [comment]
I will post notes in a few days on the expected behavior when a type alias is involved, in case you or others have feedback.

**Comment by nishanths** [comment]
I've created a branch `typealias` and a pull request (#26) that defines/implements exhaustiveness for type-aliased enums. You can try it out by running:

```
go install github.com/nishanths/exhaustive/cmd/exhaustive@typealias
```

Below are the relevant parts from the documentation defining the behavior. Any feedback is welcome. The full doc can be found in [doc.go](https://github.com/nishanths/exhaustive/blob/typealias/doc.go) in the pull request branch. 

The testdata files [typealias/typealias.go](https://github.com/nishanths/exhaustive/blob/typealias/testdata/src/typealias/typealias.go) and [enum/typealias/typealias.go](https://github.com/nishanths/exhaustive/blob/typealias/testdata/src/enum/typealias/typealias.go) have, what I think, is a thorough test of all the possibilities related to type aliases.

## Documentation

### Definition of enum

...

For a constant to be an enum member, it must be declared in the same scope as
the enum type. This implies that all members of an enum type must be present
in the same package as the enum type (if a constant of the enum type is defined
in a package different from the enum type's package, the constant will not
constitute an enum member).

...

### Type aliases and exhaustiveness

The type alias proposal says that in a type alias declaration:

    type T1 = T2

T1 is merely an alternate spelling for T2, and nearly all analysis of code
involving T1 proceeds by first expanding T1 to T2 [*]. For the exhaustive
analyzer, this means that a switch statement that switches on a value of type T1
is in effect switching on a value of type T2.

Consider the type alias declaration type T1 = T2, where T2 or its underlying
type is an enum type (i.e. a named type whose underlying type is integer, float,
or string; for details see section 'Definition of enum'). A switch statement
that switches on a value of type T1 is exhaustive if all of the enum type T2's
members are listed in the switch statement cases. It is worth highlighting that
only constants declared in the same package as the type T2 can constitute T2's
enum members (as defined in section 'Definition of enum').

**Comment by nishanths** [comment]
I've merged the PR, and will tag v0.7.0 including this change. Please create an issue if you notice a bug or have feedback. Thanks!

**Comment by nishanths** [comment]
Latest docs: https://godocs.io/github.com/nishanths/exhaustive#hdr-Type_aliases
